### PR TITLE
Support Fedora

### PR DESCRIPTION
**Pull request by eidins**
Replacement for https://github.com/open-eid/DigiDoc4-Client/pull/935 that fixes the conflict

### CODE REVIEW

**Comment by eidins** [comment]
Works with package output from https://github.com/open-eid/libdigidocpp/pull/441

File: CMakeLists.txt [hunk 1]
```diff
@@ -22,7 +22,30 @@ set_env( MOBILEID_URL "https://dd-mid.ria.ee/mid-api" CACHE STRING "URL for Mobi
...
-
+set(CPACK_PACKAGE_VERSION ${VERSION})
+set(CPACK_GENERATOR "RPM")
+set(CPACK_PACKAGE_NAME "qdigidoc4")
+set(CPACK_PACKAGE_RELEASE 1)
+set(CPACK_PACKAGE_CONTACT "Andrus Randveer")
+set(CPACK_PACKAGE_VENDOR "RIA")
+set(CPACK_PACKAGING_INSTALL_PREFIX "/usr/")
+set(CPACK_RPM_EXCLUDE_FROM_AUTO_FILELIST_ADDITION
+        "/usr/local" "/usr/local/etc" "/usr/share/man"
```
Comment (metsma) [inline CMakeLists.txt:33]: Package should not install anything into /usr/local folder

Comment (eidins) [reply CMakeLists.txt]: It had some conflicts with the Fedora 32 (maybe 33?) filesystem package, but in 34 it's fine.

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -90,6 +90,40 @@ jobs:
...
+      run: |
+        dnf install -y \
+          cmake openssl-devel xerces-c-devel \
+          xml-security-c-devel zlib-devel vim-common \
+          https://www.codesynthesis.com/download/xsd/4.0/linux-gnu/x86_64/xsd-4.0.0-1.x86_64.rpm
+    - name: Build
+      run: |
+        mkdir build
+        cd build
+        cmake ..
```
Comment (eidins) [inline .github/workflows/build.yml:119]:
```suggestion
        cmake -DCMAKE_INSTALL_SYSCONFDIR=/etc ..
```
@metsma Should I do this here too?

File: .github/workflows/build.yml [hunk 3]
```diff
@@ -90,6 +90,40 @@ jobs:
...
+      with:
+        submodules: recursive
+    - name: Install Build Deps
+      run: dnf groupinstall -y "Development Tools" "Development Libraries" fedora-packager
+    - name: Install Deps
+      run: |
+        dnf install -y \
+          cmake openssl-devel xerces-c-devel \
+          xml-security-c-devel zlib-devel vim-common \
+          https://www.codesynthesis.com/download/xsd/4.0/linux-gnu/x86_64/xsd-4.0.0-1.x86_64.rpm
```
Comment (metsma) [inline .github/workflows/build.yml:114]: You need pull libdigidocpp.rpm artifact. Look Ubuntu build.
And probably install qt development environment

File: .github/workflows/build.yml (hunk 2 above)
Comment (metsma) [reply .github/workflows/build.yml:119]: I don't think digidoc installs anything under etc

**Review by metsma — CHANGES_REQUESTED** [review]

**Comment by eidins** [comment]
@metsma I think this works now

File: .github/workflows/build.yml [hunk 4]
```diff
@@ -90,6 +90,53 @@ jobs:
...
+      uses: actions/checkout@v1
+      with:
+        submodules: recursive
+    - name: Install Build Deps
+      run: dnf groupinstall -y "Development Tools" "Development Libraries" fedora-packager
+    - name: Install Deps
+      run: |
+        dnf install -y \
+          cmake openssl-devel xerces-c-devel \
+          xml-security-c-devel zlib-devel vim-common \
```
Comment (metsma) [inline .github/workflows/build.yml:113]: You don't need xerces/xml-security-c/zlib/xsd dependencies here

File: .github/workflows/build.yml [hunk 5]
```diff
@@ -90,6 +90,50 @@ jobs:
...
+    - name: Checkout
+      uses: actions/checkout@v1
+      with:
+        submodules: recursive
+    - name: Install Build Deps
+      run: dnf groupinstall -y "Development Tools" "Development Libraries" fedora-packager
+    - name: Install Deps
+      run: |
+        dnf install -y \
+          cmake openssl-devel vim-common qt5-qtsvg-devel
```
Comment (metsma) [inline .github/workflows/build.yml]: Could not find a package configuration file provided by "Qt5LinguistTools"  with any of the following names:
    Qt5LinguistToolsConfig.cmake
I think you need install also install qt5-linguist package

File: .github/workflows/build.yml (hunk 2 above)
Comment (metsma) [reply .github/workflows/build.yml:129]:
```suggestion
        cmake -DCMAKE_INSTALL_PREFIX=/usr ..
```
Can you add for now to see if it fixes
CMake Error: failed to create symbolic link '/__w/DigiDoc4-Client/DigiDoc4-Client/build/_CPack_Packages/Linux/RPM/qdigidoc4-4.2.11.53-1.x86_64//usr/local/share/icons/hicolor/16x16/mimetypes/application-vnd.etsi.asic-s+zip.png': no such file or directory

File: README.md [hunk 6]
```diff
@@ -23,7 +23,10 @@
 
    * Install
 
+         # Ubuntu
          sudo apt install cmake qttools5-dev libqt5svg5-dev qttools5-dev-tools libpcsclite-dev libssl-dev libdigidocpp-dev libldap2-dev gettext pkg-config
+         # Fedora
+         sudo dnf install qt5-qttools-devel qt5-qtsvg-devel pcsc-lite-devel openssl-devel libdigidocpp openldap-devel gettext pkg-config
```
Comment (metsma) [inline README.md:29]:
```suggestion
         sudo dnf install qt5-qttools-devel qt5-qtsvg-devel qt5-linguist pcsc-lite-devel openssl-devel libdigidocpp openldap-devel gettext pkg-config
```

**Review by metsma — CHANGES_REQUESTED** [review]

**Review by metsma — APPROVED** [review]

**Pull request merged**